### ISSUE TITLE
None passed in result list from DeferredList callback

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11406]
|<img alt="ppetru's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ppetru reported|
|-|-|
|Trac ID|trac#362|
|Type|defect|
|Created|2003-10-21 01:04:17Z|

Attachments:

* [test.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/8d6/8d6f9131366dac0c298ee725e6577d6e0a54e832/a36e355ddb203d7d4133221f339dc406cb9f480f.py) (501 bytes) - added by ppetru on 2003-10-21 01:04:17Z - 
* [test2.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/8d6/8d6f9131366dac0c298ee725e6577d6e0a54e832/2a41da11c987266d7c39b70a05e3cc3cba268c18.py) (453 bytes) - added by ppetru on 2003-10-21 01:39:38Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__362 362
type__defect defect
reporter__ppetru ppetru
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1066698257000000 1066698257000000
changetime__1469005792464745 1469005792464745
version__ 
owner__ 
cc__ppetru
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="ppetru's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ppetru commented|
|-|-|

```
#!html
<pre>
The attached test, when run on my machine (Debian/sid, twisted deb packages 1.0.
7+1.1.0alpha3-1) produces the following output:

[(1, 'one'), None, (1, 'three')]

The result should be

[(1, 'one'), (2, 'two'), (1, 'three')]

... unless I'm missing something.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="ppetru's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ppetru commented|
|-|-|

```
#!html
<pre>
I made a typo in the expected result, the second item in the list should of 
course be (1, 'two')
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="ppetru's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|ppetru commented|
|-|-|

```
#!html
<pre>
Another interesting fact: if the deferreds are passed to the DeferredList 
constructor (as in the attached test2.py file), it runs at expected.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
That's not a bug, unfortunately, that's a feature.
Once the deferred list has reached a "full" state (in your case,
immediately, since it has no deferreds to wait for).
I have noted the subtleties of addDeferred in the deferred howto.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub commented|
|-|-|

In changeset 620a3bfeb7034b6aa7cf36fc46d02c90d48afa60
```
#!CommitTicketReference repository="" revision="620a3bfeb7034b6aa7cf36fc46d02c90d48afa60"
Merge pull request #362 from twisted/8629-scop-pypywsgi

Author: scop
Reviewer: rodrigc
Fixes: #8629

Fix test_wsgi.StartResponseTests on PyPy
```
